Jenkinsfile

```diff
@@ -16,44 +16,48 @@ def surefireTestReports='target/surefire-reports/TEST-*.xml'
 timestamps {
   node {
     ansicolor {
-       try {
-         pullRequests.ensureJobDescription()
-         stage('Checkout') {
-          info.printNode()
-          notify.started()
-          checkout scm
-        }
+      // ensure the build can handle at-signs in paths:
+      dir("@") {
+        try {
+          pullRequests.ensureJobDescription()
+          stage('Checkout') {
+            info.printNode()
+            notify.started()
+            checkout scm
+          }
 
-        stage('Check build tools') {
-          info.printNode()
+          stage('Check build tools') {
+            info.printNode()
 
-          // Note: if next stage happens on another node, mvnw might have to download again
-          sh "./mvnw --version"
-        }
+            // Note: if next stage happens on another node, mvnw might have to download again
+            sh "./mvnw --version"
+          }
 
-        stage('Build') {
-          info.printNode()
-          info.printEnv()
-          sh """./mvnw -e clean verify \
-                     --batch-mode \
-                     --settings .travis-settings.xml \
-                     --update-snapshots \
-                     -Dmaven.test.failure.ignore \
-                     -DstaticAnalysis \
-             """
-             notify.successful()
+          stage('Build') {
+            info.printNode()
+            info.printEnv()
+            sh """./mvnw -e clean verify \
+                --batch-mode \
+                --settings .travis-settings.xml \
+                --update-snapshots \
+                -Dmaven.test.failure.ignore \
+                -DstaticAnalysis \
+            """
+            // step([$class: 'CheckStylePublisher', pattern: '**/target/checkstyle-result.xml', unstableTotalAll:'0'])
+            // step([$class: 'PmdPublisher', pattern: '**/target/pmd.xml', unstableTotalAll:'0'])
+            // step([$class: 'FindBugsPublisher', pattern: '**/findbugsXml.xml', unstableTotalAll:'0'])
+            junit testResults: "**/${surefireTestReports}"
```

Comment:
You mean, after mvn exits with non-zero?

In this case, if mvn fails (eg can't compile), an exception will be thrown, and the junit step won't be reached.